src/main/java/com/github/khanmurtuza/EmailScraper.java

```diff
@@ -0,0 +1,130 @@
+        final Set<String> allEmails = new HashSet<String>();
+
+        addLinkToCrawler(inputUrl);
+
+        while (!linksToCrawl.isEmpty()) {
+            final String linkToCrawl = linksToCrawl.poll();
+
+            try {
+                final Document document = crawler.getDocument(linkToCrawl);
+
+                // add emails found in current page
+                allEmails.addAll(getEmails(document));
+
+                // get all unique links found in current page
+                addLinksToCrawler(getLinks(document));
+            } catch (IOException e) {
+                // continue crawling when there is exception connecting to a url
+                continue;
+            }
+        }
+        return allEmails;
+    }
+
+    private void addLinkToCrawler(final String link) {
+        if (!linksTracker.contains(link)) {
+            linksToCrawl.offer(link);
+            linksTracker.add(link);
+        }
+    }
+
+    private void addLinksToCrawler(final Set<String> links) {
+        for (final String link : links) {
+            addLinkToCrawler(link);
+        }
+    }
+
+    private Set<String> getEmails(final Document document) {
+        final Set<String> emails = new HashSet<String>();
+        final Pattern pattern = Pattern.compile(EMAIL_REGX);
+        final Matcher matcher = pattern.matcher(document.text());
+        while (matcher.find()) {
+            emails.add(matcher.group());
+        }
+
+        return emails;
+    }
+
+    private Set<String> getLinks(final Document document) {
+        final Set<String> links = new HashSet<String>();
+        final Elements elements = document.select("a[href]");
+        for (Element elem : elements) {
+            final String absUrl = crawler.getAbsUrl(elem);
+            if (!StringUtils.isBlank(absUrl)) {
+                if (crawler.shouldCrawl(absUrl)) {
+                    links.add(absUrl);
+                }
+            }
+        }
+        return links;
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length <= 0) {
+            System.out.println("Please provide url to crawl");
+            System.exit(1);
+        }
+
+        final String input = args[0];
+
+        String formattedInputUrl = null;
+        if (!input.startsWith(SCHEMA_HTTP) && !input.startsWith(SCHEMA_HTTPS)) {
+            formattedInputUrl = SCHEMA_HTTP + "://" + input;
+        } else {
+            formattedInputUrl = input;
+        }
+
+        final UrlValidator urlValidator = new UrlValidator(SUPPORTED_SCHEMAS);
+        if (!urlValidator.isValid(formattedInputUrl)) {
+            System.out.println("Given url is not valid: " + input);
+            System.exit(1);
+        }
+
+        final EmailScraper emailScraper = new EmailScraper(new WebsiteCrawler(formattedInputUrl));
+        for (final String link : emailScraper.getAllEmails(formattedInputUrl)) {
+            System.out.println(link);
```

Comment:
I could add debug print statement in EmailScraper#getAllEmails, but ideally this should be achieved by sending message to EmailScraper user as emails are collected